api/py/ai/chronon/repo/validator.py

```diff
@@ -110,7 +109,29 @@ def get_pre_derived_join_columns(join: Join) -> List[str]:
             prefix = jp.prefix + "_" if jp.prefix else ""
             gb_prefix = jp.groupBy.metaData.name.replace(".", "_")
             output_columns.append(prefix + gb_prefix + "_" + col)
-    return output_columns
+    return output_columns + get_external_columns(join)
```

Comment:
nit: refactor the output_columns into a separate method e.g. `get_all_group_by_columns` and change this into

```
return get_all_group_by_columns(join) + get_all_external_columns(join)
```